lang/ruby/lib/avro/schema.rb

```diff
@@ -137,6 +137,18 @@ def initialize(type)
     # Deprecated in favor of {#type_sym}.
     def type; @type_sym.to_s; end
 
+    # Returns the MD5 fingerprint of the schema as an Integer.
+    def md5_fingerprint
```

Comment:
this would be a nice optimization, but I think that can be a follow on. Interested in adding it @djudd ?